scripts/install-build-tools.sh

```diff
@@ -14,30 +15,218 @@ if (( $EUID != 0 )); then
+echo "Python3 versions supported: ${PYTHON_VERSIONS[@]}"
+
+# make a virtual environement to install python packages
+create_venv_install_python() {
+  PY_LOC=$1
+  if [ -z "$PY_LOC" ]; then
+    echo "Python path not provided"
+    exit 1
+  fi
+  PY_VERSION=$2
+  if [ -z "$PY_VERSION" ]; then
+    echo "python version not provided"
+    exit 1
+  fi
+  # remove existing virtual environment if it exists - global var
+  ENV_LOCATION="${ROOT}/${ENV_NAME}"
+  if [ -d "${ENV_LOCATION}" ]; then
+    if rm -rf "${ENV_LOCATION}"; then
+      echo "Removed existing virtual environment"
+    else
+      echo "Failed to remove existing virtual environment"; exit 1
+    fi
+  fi
+  # install pip for linux
+  if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    if ! $SUDO apt install -y python3-pip; then
+      echo "Failed to install python3-pip"
+    fi
+    if ! $SUDO python${PY_VERSION} -m pip install --upgrade pip; then
+      echo "Failed to upgrade pip"
+    fi
+  fi
+  # install python3 venv module for linux
+  if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    if ! $SUDO apt install -y "python${PY_VERSION}-venv"; then
+      echo "Failed to install python${PY_VERSION}-venv"; exit 1
+    else
+      echo "Installed python${PY_VERSION}-venv"
+    fi
+  fi
+  # create virtual environment with specific python version
+  if "${PY_LOC}" -m venv "${ENV_NAME}"; then
+    echo "Virtual environment '${ENV_NAME}' created"
+  else
+    echo "Virtual environment creation failed"; exit 1
+  fi
+  # activate virtual environment
+  if source "${ENV_NAME}/bin/activate"; then
+    echo "Virtual environment '${ENV_NAME}' activated"
+  else
+    echo "Virtual environment activation failed"; exit 1
+  fi
+  # install python packages
+  if "${ENV_NAME}/bin/pip" install -r requirements.txt; then
+    echo "Success installing python packages"
+  else
+    echo "Failure installing python packages"
+    deactivate
+    exit 1
+  fi
+  deactivate
+}
+
+
+echo "OS Type: $OSTYPE"
+# macOS install with homebrew
 if [[ "$OSTYPE" == "darwin"* ]]; then
+
+  # macOS does not support running shell scripts as root with homebrew
+  if [ $EUID -eq 0 ]; then
+    echo -e "Mac users should run this script without 'sudo'. Exiting..."
+    exit 1
+  fi
+
   CPUS=$(sysctl -n hw.ncpu)
   # ensure development environment is set correctly for clang
   $SUDO xcode-select -switch /Library/Developer/CommandLineTools
-  brew install llvm@14 googletest google-benchmark lcov make wget cmake curl
+
+  if ! brew --version &>/dev/null; then
+    echo -e "${cyan}Homebrew is required to install dependencies.${end}"
+    exit 1
+  fi
+
+  brew install llvm@14 googletest google-benchmark lcov make wget cmake bash bc
+  brew upgrade bash
+
+  BREW_ROOT=$(brew --prefix)
+
   CLANG_TIDY=/usr/local/bin/clang-tidy
   if [ ! -L "$CLANG_TIDY" ]; then
-    $SUDO ln -s $(brew --prefix)/opt/llvm@14/bin/clang-tidy /usr/local/bin/clang-tidy
+    $SUDO ln -s "${BREW_ROOT}/opt/llvm@14/bin/clang-tidy /usr/local/bin/clang-tidy"
```

Comment:
Quoting in this fashion passes only one path to `ln` which almost certainly doesn't exist. Probably the paths need to be individually quoted.

Reply:
mistake on my part...just fixed